app/screens/settings/advanced/index.tsx

```diff
@@ -32,26 +37,53 @@ const EMPTY_FILES: ReadDirItem[] = [];
 
 type AdvancedSettingsProps = {
     componentId: string;
-}
+};
 const AdvancedSettings = ({componentId}: AdvancedSettingsProps) => {
     const theme = useTheme();
     const intl = useIntl();
     const serverUrl = useServerUrl();
-    const [dataSize, setDataSize] = useState<number|undefined>(0);
+    const [dataSize, setDataSize] = useState<number | undefined>(0);
     const [files, setFiles] = useState<ReadDirItem[]>(EMPTY_FILES);
     const styles = getStyleSheet(theme);
 
-    const getAllCachedFiles = async () => {
-        const {totalSize, files: cachedFiles} = await getAllFilesInCachesDirectory(serverUrl);
+    const getAllCachedFiles = useCallback(async () => {
+        const {totalSize = 0, files: cachedFiles} =
+            await getAllFilesInCachesDirectory(serverUrl);
         setDataSize(totalSize);
         setFiles(cachedFiles || EMPTY_FILES);
-    };
+    }, [serverUrl]);
 
     const onPressDeleteData = preventDoubleTap(async () => {
         try {
             if (files.length > 0) {
-                await deleteFileCache(serverUrl);
-                await getAllCachedFiles();
+                const {formatMessage} = intl;
+
+                Alert.alert(
+                    formatMessage({
+                        id: t('advanced_settings.delete_data'),
+                        defaultMessage: 'Delete Documents & Data',
+                    }),
+                    formatMessage({
+                        id: t('mobile.advanced_settings.delete_message.confirmation'),
```

Comment:
previous localization id does not have the prefix `modal`

Reply:
corrected